core/thread/each_caller_location_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative '../../spec_helper'
+
+describe "Thread.each_caller_location" do
+  ruby_version_is "3.2" do
+    it "iterates through the current execution stack and matches caller_locations content and type" do
+      ScratchPad.record []
+      Thread.each_caller_location { |l| ScratchPad << l; }
+
+      ScratchPad.recorded.map(&:to_s).should == caller_locations.map(&:to_s)
+      ScratchPad.recorded[0].should be_kind_of(Thread::Backtrace::Location)
+    end
+
+    it "returns subset of 'Thread.to_enum(:each_caller_location)' locations" do
+      i = 0
+      ar = []
+      ecl = Thread.each_caller_location do |x|
+        ar << x
+        i += 1
+        break x if i == 1
```

Comment:
**minor:** we don't need this skipping if we would like to check all the returned stack locations.